### PR TITLE
Gulp task runner

### PR DESCRIPTION
**Pull request by honzaskovran**
Grunt replaced by gulp with support of ES6. Added option to choose css preprocessor (sass/less)

### CODE REVIEW

File: bin/deployment/init-project.php [hunk 1]
```diff
@@ -1,37 +1,29 @@
 <?php
 use Instante\Deployment\ProjectInitializer;
-
```
Comment (richard-ejem) [inline bin/deployment/init-project.php:3]: Please return the empty lines back :)

File: bin/deployment/ProjectInitializer.php [hunk 2]
```diff
@@ -224,8 +261,28 @@ private function deployIndex()
 
     public function removeItself()
     {
-        unlink(__DIR__ . '/ProjectInitializer.php');
-        unlink(__DIR__ . '/init-project.php');
-        unlink(__DIR__ . '/templates/init.latte');
+        //unlink(__DIR__ . '/ProjectInitializer.php');
```
Comment (richard-ejem) [inline bin/deployment/ProjectInitializer.php]: why? project initializer should remove itself after initialize.

File: frontend/package.json [hunk 3]
```diff
@@ -1,31 +1,31 @@
 {
-    "name": "YourAppName",
```
Comment (richard-ejem) [inline frontend/package.json:2]: Please return to 4-space indentation

File: bin/deployment/ProjectInitializer.php [hunk 4]
```diff
@@ -77,6 +79,40 @@ public function setAuthor($name, $email)
...
 
+    /**
+     * @param string $preprocessor
+     * @return $this
+     */
+    public function setCssPreprocessor($preprocessor) {
+        switch($preprocessor) {
+            case 'sass':
+                $this->bootstrap = ['name' => 'bootstrap-sass', 'version' => '3.3.7'];
+                $this->removeDirecory($this->dir . '/frontend/src/less');
```
Comment (richard-ejem) [inline bin/deployment/ProjectInitializer.php]: The actual work should be done in initialize() method. This setter should only prepare configuration in ProjectInitializer attributes.
Please add SASS/LESS choice to initializeFromConsole() method too.

File: bin/deployment/ProjectInitializer.php [hunk 5]
```diff
@@ -77,6 +79,40 @@ public function setAuthor($name, $email)
...
+                break;
+            case 'less':
+                $this->bootstrap = ['name' => 'bootstrap', 'version' => '3.3.4'];
+                $this->removeDirecory($this->dir . '/frontend/src/sass');
```
Comment (richard-ejem) [inline bin/deployment/ProjectInitializer.php]: dtto should happen in initialize()

File: bin/deployment/ProjectInitializer.php [hunk 6]
```diff
@@ -77,6 +79,40 @@ public function setAuthor($name, $email)
...
+                break;
+            default:
+                $this->errors[] = 'Bootstrap not set';
+        }
+        $this->configureGulpfile($preprocessor);
```
Comment (richard-ejem) [inline bin/deployment/ProjectInitializer.php]: dtto should happen in initialize()

**Review by richard-ejem — CHANGES_REQUESTED** [review]

**Comment by honzaskovran** [comment]
Done, please recheck

**Comment by honzaskovran** [comment]
Travis is now passing. I also added test for gulpfile configuration

**Review by richard-ejem — APPROVED** [review]

**Pull request merged**